### PR TITLE
ARTEMIS-3901 Fix artemis perf client --durable fails on exception

### PR DESCRIPTION
**Pull request by jsmucr**
https://issues.apache.org/jira/browse/ARTEMIS-3901

### CODE REVIEW

File: artemis-cli/src/main/java/org/apache/activemq/artemis/cli/commands/messages/perf/PerfClientCommand.java [hunk 1]
```diff
@@ -131,6 +134,7 @@ protected void onExecuteBenchmark(final ConnectionFactory producerConnectionFact
          .setTimeProvider(() -> TimeUnit.NANOSECONDS.toMicros(System.nanoTime()))
          .setCanDelayMessageCount(true)
          .setSharedSubscription(sharedSubscription)
+         .setClientID(UUID.randomUUID().toString())
```
Comment (gtully) [inline artemis-cli/src/main/java/org/apache/activemq/artemis/cli/commands/messages/perf/PerfClientCommand.java:137]: this seems odd, maybe durableSubscription boolean needs to be an identity in the form of a string. If the generated uuid is the identity of the subscription, then it won't be possible to access those messages on a second attempt.

Comment (jsmucr) [reply artemis-cli/src/main/java/org/apache/activemq/artemis/cli/commands/messages/perf/PerfClientCommand.java:137]: IIRC there was an exception which requested the client ID to be set. So I just tried doing this and it worked. I'm building now so I can't check. I'll be back. :-)

Comment (jsmucr) [reply artemis-cli/src/main/java/org/apache/activemq/artemis/cli/commands/messages/perf/PerfClientCommand.java:137]: There it is:
```
java.lang.RuntimeException: javax.jms.IllegalStateException: Cannot create durable subscription - client ID has not been set

	at org.apache.activemq.artemis.cli.commands.messages.perf.MessageListenerBenchmark.start(MessageListenerBenchmark.java:221)
	at org.apache.activemq.artemis.cli.commands.messages.perf.PerfClientCommand.onExecuteBenchmark(PerfClientCommand.java:171)
	at org.apache.activemq.artemis.cli.commands.messages.perf.PerfCommand.execute(PerfCommand.java:79)
	at org.apache.activemq.artemis.cli.commands.messages.perf.PerfClientCommand.execute(PerfClientCommand.java:214)
	at org.apache.activemq.cli.test.CliPerfClientTest$1.execute(CliPerfClientTest.java:55)
	at org.apache.activemq.cli.test.CliPerfClientTest.start(CliPerfClientTest.java:58)
	at org.apache.activemq.cli.test.CliPerfClientTest.testDurableStarts(CliPerfClientTest.java:68)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
Caused by: javax.jms.IllegalStateException: Cannot create durable subscription - client ID has not been set
	at org.apache.activemq.artemis.jms.client.ActiveMQSession.createConsumer(ActiveMQSession.java:865)
	at org.apache.activemq.artemis.jms.client.ActiveMQSession.createDurableConsumer(ActiveMQSession.java:661)
	at org.apache.activemq.artemis.jms.client.ActiveMQSession.createDurableConsumer(ActiveMQSession.java:643)
	at org.apache.activemq.artemis.cli.commands.messages.perf.MessageListenerBenchmark.start(MessageListenerBenchmark.java:214)
	... 37 more
```

Comment (gtully) [reply artemis-cli/src/main/java/org/apache/activemq/artemis/cli/commands/messages/perf/PerfClientCommand.java:137]: it does need to be set, but it must be stable. a durable sub queue is identified by the clientId and subscriptionName pair, those need to be stable across connections such that a consumer gets back to the same underlying queue.
typically the user provides both on the command line.

Comment (jsmucr) [reply artemis-cli/src/main/java/org/apache/activemq/artemis/cli/commands/messages/perf/PerfClientCommand.java:137]: Aah, yes, I missed the line above:
```
         .setClientID(getClientID())
```

Comment (jsmucr) [reply artemis-cli/src/main/java/org/apache/activemq/artemis/cli/commands/messages/perf/PerfClientCommand.java]: Thank you for review.

**Comment by clebertsuconic** [comment]
I added a tweak and merged it

**Pull request merged**